async_fuse/src/memfs/fs_util.rs

```diff
@@ -289,20 +289,20 @@ pub async fn load_dir_data(dirfd: RawFd) -> anyhow::Result<BTreeMap<OsString, Di
 }
 
 /// Helper function to load file data
-pub async fn load_file_data(fd: RawFd, file_size: usize) -> anyhow::Result<Vec<u8>> {
+pub async fn load_file_data(fd: RawFd, offset: usize, len: usize) -> anyhow::Result<Vec<u8>> {
     let file_data_vec = smol::unblock(move || {
-        let mut file_data_vec: Vec<u8> = Vec::with_capacity(file_size);
+        let mut file_data_vec: Vec<u8> = Vec::with_capacity(len);
         unsafe {
             file_data_vec.set_len(file_data_vec.capacity());
         }
```

Comment:
Potential unsoundness here.

See <https://doc.rust-lang.org/std/vec/struct.Vec.html#safety-2>

Reply:
Yes, I guess you mentioned it before. Let me set the length after loading the file, just like what they do in the example.

Reply:
The simplest answer is `vec![0; len]`.

Reply:
The initialized is not necessary.